src/clib/pio_file.c

```diff
@@ -283,6 +283,10 @@ PIOc_closefile(int ncid)
             ierr = ncmpi_close(file->fh);
             break;
 #endif
+	case PIO_IOTYPE_GDAL:
```

Comment:
I think that you need a line like 274 here:
if (ios->io_rank == 0)
this should only be called on the root io task.